### PR TITLE
Add a more complex heatmap example

### PR DESCRIPTION
**Pull request by mcnuttandrew**
screen shot 2018-07-17 at 7 37 44 pm

This example addresses questions in #849 #835 #790

### CODE REVIEW

File: showcase/plot/labeled-heatmap.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+import {scaleLinear} from 'd3-scale';
+
+import {XYPlot, XAxis, YAxis, HeatmapSeries, LabelSeries} from 'index';
+
+const alphabet = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J'];
+const data = [...new Array(alphabet.length)].reduce((acc, _, idx) => {
```
Comment (benshope) [inline showcase/plot/labeled-heatmap.js]: what's with creating `[...new Array(alphabet.length)]` when you could just iterate over `alphabet`?

File: showcase/plot/labeled-heatmap.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  return acc.concat([...new Array(alphabet.length)].map((d, jdx) => {
+    return {
+      x: `${alphabet[idx]}1`,
+      y: `${alphabet[jdx]}2`,
+      color: (idx + jdx) % Math.floor(jdx / idx) || idx
+    };
+  }));
+}, []);
+const {min, max} = data.reduce((acc, row) => {
+  return {
```
Comment (benshope) [inline showcase/plot/labeled-heatmap.js]: `() => ({})` instead of `() => { return {}; }`

File: showcase/plot/labeled-heatmap.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import React from 'react';
+import {scaleLinear} from 'd3-scale';
+
+import {XYPlot, XAxis, YAxis, HeatmapSeries, LabelSeries} from 'index';
+
+const alphabet = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J'];
+const data = [...new Array(alphabet.length)].reduce((acc, _, idx) => {
+  return acc.concat([...new Array(alphabet.length)].map((d, jdx) => {
+    return {
```
Comment (benshope) [inline showcase/plot/labeled-heatmap.js]: `() => ({})` instead of `() => { return {}; }`

File: showcase/plot/labeled-heatmap.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }));
+}, []);
+const {min, max} = data.reduce((acc, row) => {
+  return {
+    min: Math.min(acc.min, row.color),
+    max: Math.max(acc.max, row.color)
+  };
+}, {min: Infinity, max: -Infinity});
+
+export default class LabeledHeatmap extends React.Component {
```
Comment (benshope) [inline showcase/plot/labeled-heatmap.js]: make stateless?

File: showcase/plot/labeled-heatmap.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+              ry: 10
+            }
+          }}
+          className="heatmap-series-example"
+          data={data}/>
+        <LabelSeries
+          data={data}
+          labelAnchorX="middle"
+          labelAnchorY="baseline"
+          getLabel={d => {
```
Comment (benshope) [inline showcase/plot/labeled-heatmap.js]: `() => ({})` instead of `() => { return {}; }`

**Comment by mcnuttandrew** [comment]
@benshope what a reasonable set of reviews! Thanks for reading so many things all at once

**Pull request merged**